### PR TITLE
[Docs] Render mermaid graph code blocks

### PR DESCRIPTION
**Pull request by SaschaMann**
Codeblocks with the language set to mermaid will be rendered as mermaid graphs.

For example, 

    ```mermaid
    graph TD
    A((A)) --> robots
    subgraph Multiple Dispatch
	    robots --> encounters
	    robots -.-> abstract-types
	    abstract-types[abstract types] -.-> modules
	    modules -.-> encounters
	    encounters -.-> extension[extending/glueing together modules]
    end
    extension --> B
    B --> performance
    performance --> Z((Z))
    ```

is rendered as the following graph: https://user-images.githubusercontent.com/20866761/73282965-abe4a180-41f2-11ea-8968-54faaaf7c91e.png

This can be quite useful when visualising and planning the progression. Unfortunately, it requires adding a dependency/external script to the website.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -22,7 +25,22 @@
...
+              return (
+                '<div class="mermaid">' + mermaid.render('mermaid-svg-' + num++, code) + "</div>"
+              );
+            }
+            return this.origin.code.apply(this, arguments);
+          }
+        }
+      }
     }
+    var num = 0;
```
Comment (iHiD) [inline index.html]: Is this needed?

Comment (SaschaMann) [reply index.html]: It's needed for the name of the svg that mermaid renders. Though I guess I could replace it with some random function if that's better.

Comment (iHiD) [reply index.html]: @SleeplessByte How do we make this less global/leaky?

Comment (ErikSchierboom) [reply index.html]: This depends a bit on whether the variable needs to be defined only here or globally at some later time. If it is the former, would an [IFFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) work?

Comment (iHiD) [reply index.html]: That sounds sensible to me.

I'm just wary of it conflicting with some other JS we add later.

File: languages/julia/reference/progression.md [hunk 2]
```diff
@@ -68,4 +68,4 @@ I think of closely coupled concepts/exercises as a “block” in the progressio
 ### OffsetArrays
 
 - Some exercise that shows that indexing is arbitrary in Julia, even if it defaults to 1-based indices.
-- Also: How to write generic code that works regardless of index (see https://docs.julialang.org/en/v1/devdocs/offset-arrays/#Generalizing-existing-code-1)
\ No newline at end of file
+- Also: How to write generic code that works regardless of index (see https://docs.julialang.org/en/v1/devdocs/offset-arrays/#Generalizing-existing-code-1)
```
Comment (SaschaMann) [inline languages/julia/reference/progression.md]: I don't know why this is highlighted as a change.

File: index.html (hunk 1 above)
Comment (SaschaMann) [reply index.html]: Replaced it with a random ID.

Comment (ErikSchierboom) [reply index.html]: Excellent!

**Comment by SaschaMann** [comment]
Added docs on how to use it

File: docs/maintainers/README.md [hunk 3]
```diff
@@ -28,6 +28,106 @@ The final step to preparing your language to accept contributions is to provide
 
 We recommend choosing a Concept that involves working with basic types, like strings and numbers. Having chosen a Concept, look for the Concept's description in this reference part of this repository. This file should have references to all tracks that have implemented the exercise. You can use these existing implementations as a starting point for your first concept exercise, which you can then tailor to your specific language.
 
+### Planning the track structure
```
Comment (ErikSchierboom) [inline docs/maintainers/README.md]: Maybe do:

```suggestion
### Visualizing the track structure
```

as the content seems to deal with the visualization aspect, not on the actual planning. Just a suggestion.

File: docs/maintainers/README.md [hunk 4]
```diff
@@ -28,6 +28,106 @@ The final step to preparing your language to accept contributions is to provide
...
+```mermaid
+    graph LR
+        basic-numbers --> advanced-numbers
+```
+
+Please refer to the [mermaid documentation](https://mermaid-js.github.io/mermaid/#/flowchart) for detailed instructions. Below you can find a few examples of basic mermaid syntax to get you started. The Julia track's [progression.md](./../../languages/julia/reference/progression.md) can also be used as a reference.
+
+*Note: to keep the page short, the example use left-to-right graphs (`LR`). For a track progression path you want to replace `graph LR` with `graph TD` for a top-to-bottom graph.*
+
+These will only be rendered on the website or in a markdown editor that supports mermaid rendering, e.g. [Typora](https://www.typora.io/). See [website.md](./website.md) for instructions to run the website locally.
```
Comment (ErikSchierboom) [inline docs/maintainers/README.md]: Maybe add link to the website?

File: docs/maintainers/README.md [hunk 5]
```diff
@@ -28,6 +28,106 @@ The final step to preparing your language to accept contributions is to provide
...
+        graph LR
+            basic-numbers --> advanced-numbers
+    ```
+
+```mermaid
+    graph LR
+        basic-numbers --> advanced-numbers
+```
+
+Please refer to the [mermaid documentation](https://mermaid-js.github.io/mermaid/#/flowchart) for detailed instructions. Below you can find a few examples of basic mermaid syntax to get you started. The Julia track's [progression.md](./../../languages/julia/reference/progression.md) can also be used as a reference.
```
Comment (ErikSchierboom) [inline docs/maintainers/README.md]: I'm wondering if we need the examples below, or if the reference to the Julia progression is enough. I was feeling slightly overwhelmed.

File: docs/maintainers/README.md (hunk 3 above)
Comment (SaschaMann) [reply docs/maintainers/README.md]: I want to emphasize that it's a planning tool. For visualising the track structure of implemented exercises, this is probably not all that useful and you'd want to generate a graph from the `config.json` instead.

Perhaps `### Visualizing the planned track structure`?

Comment (SaschaMann) [reply docs/maintainers/README.md]:
```suggestion
### Visualizing the planned track structure
```

**Comment by SaschaMann** [comment]
Applied your suggestions

File: docs/maintainers/README.md (hunk 3 above)
Comment (ErikSchierboom) [reply docs/maintainers/README.md]: I like that!

**Comment by ErikSchierboom** [comment]
I'll review this tomorrow!

File: index.html [hunk 6]
```diff
@@ -2,11 +2,14 @@
...
   <meta charset="UTF-8">
-  <title>Document</title>
+  <title>Exercism v3</title>
   <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
   <meta name="description" content="Description">
   <meta name="viewport" content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
   <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/docsify-themeable@0/dist/css/theme-simple.css">
+
+  <!-- Render triple backtick mermaid blocks as a mermaid graph -->
+  <script src="//unpkg.com/mermaid@8.4.6/dist/mermaid.min.js"></script>
```
Comment (iHiD) [inline index.html]: Can this move down to below `<div id="app"></div>"` pls.

Comment (SaschaMann) [reply index.html]: Do JS libraries not belong in the `<head>` typically?

Comment (ErikSchierboom) [reply index.html]: @SaschaMann It depends. The thing is that if it is in the `<head>`, it will potentially block loading the rest of the page.

Comment (iHiD) [reply index.html]: > Do JS libraries not belong in the <head> typically?

As Erik says it depends, but my strong opinion is that basically they never should. So on Exercism, all JS loads after the rest of the page - that means that the site can be rendering text, images, stylesheets, fonts, etc before the JavaScript starts downloading and kicking in. There are special directives you can use on the <script> tag to make JavaScript defer itself until that happens, but they're browser-dependent.

If you're interested in more, I think this article is one of the best out there for the current situation: https://flaviocopes.com/javascript-async-defer

Comment (SaschaMann) [reply index.html]: Thanks for explaining. I moved it.

**Comment by iHiD** [comment]
Great. Thanks. If you post a thing to #v3 channel linking to the example and how to do it, I'll `@channel` it :)

**Pull request merged**